api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForLessThanXmonthsCalcultion.java

```diff
@@ -0,0 +1,51 @@
+package org.openmrs.module.eptsreports.reporting.calculation.pvls;
+
+import org.openmrs.Concept;
+import org.openmrs.Obs;
+import org.openmrs.api.context.Context;
+import org.openmrs.calculation.patient.PatientCalculationContext;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.reporting.calculation.AbstractPatientCalculation;
+import org.openmrs.module.eptsreports.reporting.calculation.BooleanResult;
+import org.openmrs.module.eptsreports.reporting.calculation.EptsCalculations;
+import org.openmrs.module.eptsreports.reporting.utils.EptsCalculationUtils;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+import java.util.Set;
+
+public class OnArtForLessThanXmonthsCalcultion extends AbstractPatientCalculation {
+	
+	@Override
+	public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> params, PatientCalculationContext context) {
+		
+		CalculationResultMap map = new CalculationResultMap();
+		Concept viralLoadConceptuuid = Context.getConceptService().getConceptByUuid("e1d6247e-1d5f-11e0-b929-000c29ad1d07");
+		// get data inicio TARV
+		CalculationResultMap arvsInitiationDateMap = calculate(new InitialArtStartDateCalculation(), cohort, context);
+		CalculationResultMap lastVl = EptsCalculations.lastObs(viralLoadConceptuuid, cohort, context);
```

Comment:
@ningosi here we also do not seem to be restricting the location. Some patients have records for multiple locations.

Reply:
Also, are we restricting to encounter type 13? From the results we seem to be but could not find on the code.